check_duplicate_versions.py

```
@@ -0,0 +1,109 @@
+from collections import defaultdict
+from typing import Dict, List
+
+
+def get_file_hash(file_path: Path) -> str:
+    """Calculate MD5 hash of a file.
+    
+    Returns:
+        Hash string on success, None on failure (logs error)
+    """
+    try:
+        md5_hash = hashlib.md5()
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)